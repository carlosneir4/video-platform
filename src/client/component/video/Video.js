import React, { Fragment } from 'react';
import ReactPlayer from 'react-player';
import styled from 'styled-components';

const Conatiner = styled.div`
  padding: 0.3em;
  margin: 5% 15%;
  padding: 2% 7%;
  
  background: #fff;
  border: 1px solid #ccc;
  border-color: rgba(0,0,0,.2);
  -webkit-box-shadow: 0px 10px 13px -7px #000000, 5px 5px 15px 5px rgba(0,0,0,0); 
  box-shadow: 0px 10px 13px -7px #000000, 5px 5px 15px 5px rgba(0,0,0,0);

  @media (max-width: 700px) {
    font-size: 20px;
  }

  input[type=button] {
   
    font-size: 15px;
    color: #ee6e73;
    background-color: #fff0;
    border:none;
    display:inline-block;
  }


`

const Title = styled.p`
    text-align: left;
    font-size: 24px;
    font-weight: bold;
    color:black;

`
const Description = styled.p`
    text-align: left;
    font-size: 16px;
    color:#696666;
`

const UploadedDate = styled.p`
    text-align: left;
    font-size: 10px;
    color:#bdb8b8;
`
const Delete = styled.div`
    text-align: right;
`

const parserDate = (timestamp) => {
    var date = new Date(timestamp);
    var dd = String(date.getDate()).padStart(2, '0');
    var mm = String(date.getMonth() + 1).padStart(2, '0'); //January is 0!
    var yyyy = date.getFullYear();

    return mm + '/' + dd + '/' + yyyy;
}

const Video = ({ id, video, handleClick }) => {

    return (
        <Conatiner>
            <Delete><input id="delete-btn" type="button" onClick={() => handleClick(id)} value="x" title="delete" /></Delete>
            <Title id="title">{video.title}</Title>
            <Description id="description">{video.description}</Description>
            <UploadedDate id="date">Uploaded Date :{parserDate(video.createTimestamp)}</UploadedDate>
            <ReactPlayer id={id}
                url={video.url}
                width='100%'
                height='50%'
                controls={true}
                volume={1}
                progressInterval={5000}
                pip={true}
            />

        </Conatiner>
    );
};

export default Video;