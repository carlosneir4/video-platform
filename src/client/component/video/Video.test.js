import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import Video from './Video'

configure({
    adapter: new Adapter()
})

describe('Testing Component <Video />', () => {

    let component;

    const id = 'someId';
    const video = {
        title: 'Title Video',
        Description: 'Some description',
        createTimestamp: '9099029292982',
        url: 'https://www.google.com/some_path'
    };
    const handleClick = jest.fn();;

    beforeAll(() => {
        component = shallow(<Video id={id} video={video} handleClick={handleClick} />);
    });

    test('should Snapshot met', () => {
        expect(toJson(component)).toMatchSnapshot()
    });

    it('calls the onClick prop function when clicked', () => {
        expect(handleClick).not.toHaveBeenCalled();
        component.dive().find('#delete-btn').simulate('click');
        expect(handleClick).toHaveBeenCalled();
    });
});