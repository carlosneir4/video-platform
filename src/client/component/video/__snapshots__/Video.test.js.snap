// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component <Video /> should Snapshot met 1`] = `
<styled.div>
  <styled.div>
    <input
      id="delete-btn"
      onClick={[Function]}
      title="delete"
      type="button"
      value="x"
    />
  </styled.div>
  <styled.p
    id="title"
  >
    Title Video
  </styled.p>
  <styled.p
    id="description"
  />
  <styled.p
    id="date"
  >
    Uploaded Date :
    NaN/NaN/NaN
  </styled.p>
  <ReactPlayer
    config={
      Object {
        "dailymotion": Object {
          "params": Object {
            "api": 1,
            "endscreen-enable": false,
          },
        },
        "facebook": Object {
          "appId": "1309697205772819",
          "attributes": Object {},
          "playerId": null,
          "version": "v3.3",
        },
        "file": Object {
          "attributes": Object {},
          "dashVersion": "3.1.3",
          "flvVersion": "1.5.0",
          "forceAudio": false,
          "forceDASH": false,
          "forceFLV": false,
          "forceHLS": false,
          "forceVideo": false,
          "hlsOptions": Object {},
          "hlsVersion": "0.14.16",
          "tracks": Array [],
        },
        "mixcloud": Object {
          "options": Object {
            "hide_cover": 1,
          },
        },
        "soundcloud": Object {
          "options": Object {
            "buying": false,
            "download": false,
            "liking": false,
            "sharing": false,
            "show_comments": false,
            "show_playcount": false,
            "visual": true,
          },
        },
        "twitch": Object {
          "options": Object {},
          "playerId": null,
        },
        "vidyard": Object {
          "options": Object {},
        },
        "vimeo": Object {
          "playerOptions": Object {
            "autopause": false,
            "byline": false,
            "portrait": false,
            "title": false,
          },
        },
        "wistia": Object {
          "customControls": null,
          "options": Object {},
          "playerId": null,
        },
        "youtube": Object {
          "embedOptions": Object {},
          "onUnstarted": [Function],
          "playerVars": Object {
            "iv_load_policy": 3,
            "modestbranding": 1,
            "playsinline": 1,
            "rel": 0,
            "showinfo": 0,
          },
        },
      }
    }
    controls={true}
    fallback={null}
    height="50%"
    id="someId"
    light={false}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onBufferEnd={[Function]}
    onClickPreview={[Function]}
    onDisablePIP={[Function]}
    onDuration={[Function]}
    onEnablePIP={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    pip={true}
    playbackRate={1}
    playing={false}
    playsinline={false}
    previewTabIndex={0}
    progressInterval={5000}
    stopOnUnmount={true}
    style={Object {}}
    url="https://www.google.com/some_path"
    volume={1}
    width="100%"
    wrapper="div"
  />
</styled.div>
`;
